isis/src/control/objs/BundleConstraint/BundleLidarRangeConstraint.cpp

```diff
@@ -0,0 +1,440 @@
+                    "(Point Id: "
+                    + measure->parentControlPoint()->id() + ", Measure Serial:"
+                    + measure->cubeSerialNumber() + ").\n";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
+
+    // initialize computed range
+    computeRange();
+  }
+
+
+  /**
+   * Destructor
+   */
+  BundleLidarRangeConstraint::~BundleLidarRangeConstraint() {
+  }
+
+
+  /**
+   * Copy constructor
+   *
+   * Constructs a BundleLidarRangeConstraint from another.
+   *
+   * @param src Source BundleLidarRangeConstraint to copy.
+   */
+  BundleLidarRangeConstraint::BundleLidarRangeConstraint(const BundleLidarRangeConstraint &src) {
+    m_dX = src.m_dX;
+    m_dY = src.m_dY;
+    m_dZ = src.m_dZ;
+    m_rangeObserved = src.m_rangeObserved;
+    m_rangeComputed = src.m_rangeComputed;
+    m_rangeObservedSigma = src.m_rangeObservedSigma;
+    m_rangeObservedWeightSqrt = src.m_rangeObservedWeightSqrt;
+    m_adjustedSigma = src.m_adjustedSigma;
+    m_vtpv = src.m_vtpv;
+  }
+
+
+  /**
+   * Assignment operator
+   *
+   * Assigns state of this BundleLidarRangeConstraint from another.
+   *
+   * @param src Source BundleLidarRangeConstraint to assign state from.
+   *
+   * @return    BundleLidarRangeConstraint& Returns reference to this
+   *            BundleLidarRangeConstraint.
+   */
+  BundleLidarRangeConstraint &BundleLidarRangeConstraint::
+      operator=(const BundleLidarRangeConstraint &src) {
+
+    // Prevent self assignment
+    if (this != &src) {
+      m_dX = src.m_dX;
+      m_dY = src.m_dY;
+      m_dZ = src.m_dZ;
+      m_rangeObserved = src.m_rangeObserved;
+      m_rangeComputed = src.m_rangeComputed;
+      m_rangeObservedSigma = src.m_rangeObservedSigma;
+      m_rangeObservedWeightSqrt = src.m_rangeObservedWeightSqrt;
+      m_adjustedSigma = src.m_adjustedSigma;
+      m_vtpv = src.m_vtpv;
+    }
+
+    return *this;
+  }
+
+
+  /**
+   * Compute range between spacecraft & lidar point on surface given the current values of the
+   * spacecraft position & lidar point coordinates in the bundle adjustment. Steps are as follows...
+   *
+   * 1) set image to establish camera model for simultaneous measure
+   * 2) get spacecraft & lidar point coordinates in body-fixed reference system
+   * 3) compute & save dX, dY, dZ between spacecraft & lidar point
+   * 4) compute range with dX, dY, dZ
+   *
+   */
+  void BundleLidarRangeConstraint::computeRange() {
+
+    // establish camera model for simultaneous measure
+    m_simultaneousMeasure->setImage();
+
+    // get spacecraft coordinates in J2000 reference system
+    std::vector<double> camPositionJ2K
+        = m_simultaneousMeasure->camera()->instrumentPosition()->Coordinate();
+
+    // body rotation "ReferenceVector" rotates spacecraft coordinates from J2000 to body-fixed
+    std::vector<double> camPositionBodyFixed
+        = m_simultaneousMeasure->camera()->bodyRotation()->ReferenceVector(camPositionJ2K);
```

Comment:
Same question here ^^^

Reply:
Same as previous. This would error on the bodyRotation call but first you'll error in the constructor.